lib/collection/property-list.js

```diff
@@ -155,34 +160,35 @@ _.extend(PropertyList.prototype, /** @lends PropertyList.prototype */ {
      * @param context {Object} Optional context to bind the predicate to.
      */
     remove: function (predicate, context) {
-        var match; // to be used if predicate is an ID
+        var match, // to be used if predicate is an ID
+            list = this;
 
-        !context && (context = this);
+        !context && (context = list);
         // if predicate is id, then create a function to remove that from array
         if (_.isString(predicate)) {
-            (match = this.one(predicate)) && (predicate = function (item) {
+            (match = list.one(predicate)) && (predicate = function (item) {
                 return (item === match);
             });
         }
         // in case an object reference is sent, prepare it for removal
-        else if (predicate instanceof this.Type) {
+        else if (predicate instanceof list.Type) {
             match = predicate;
             predicate = function (item) {
                 return (item === match);
             };
         }
 
-        _.isFunction(predicate) && _.remove(this.members, function (item) {
+        _.isFunction(predicate) && _.remove(list.members, function (item) {
             var index;
             if (predicate.apply(context, arguments)) {
-                if ((index = item[this._postman_propertyIndexKey]) && (index = String(index))) {
-                    this._postman_propertyIndexCaseInsensitive && (index = index.toLowerCase());
-                    delete this.reference[index];
+                if ((index = item[list._postman_propertyIndexKey]) && (index = String(index))) {
+                    list._postman_propertyIndexCaseInsensitive && (index = index.toLowerCase());
+                    delete list.reference[index];
                 }
                 delete item[__PARENT]; // unlink from its parent
                 return true;
             }
-        }, this);
+        });
```

Comment:
Use a `.bind()` at this place. That is what the old code used to do, and there's no reason to change it now.